lib/graphql/persisted_queries/store_adapters/memcached_store_adapter.rb

```diff
@@ -17,11 +17,23 @@ def initialize(dalli_client:, expiration: nil, namespace: nil)
         end
 
         def fetch_query(hash)
-          @dalli_proc.call { |dalli| dalli.get(key_for(hash)) }
+          @dalli_proc.call do |dalli|
+            dalli.get(key_for(hash)).tap do |result|
+              if result
+                trace("fetch_query.cache_hit", adapter: :memcached)
+              else
+                trace("fetch_query.cache_miss", adapter: :memcached)
+              end
```

Comment:
Instead of doing something like this in every `StoreAdapter`, we could move this to the `BaseStoreAdapter` with some minor modifications and then we could avoid the duplication.

Something like this:

```ruby
class BaseStoreAdapter
  # ...

  def fetch_query(_hash)
    fetch(_hash).tap do |result|
      if result
        trace("fetch_query.cache_hit", adapter: @name)
      else
        trace("fetch_query.cache_miss", adapter: @name)
      end
    end
  end

  # ...

  private

  def fetch(_hash)
    raise NotImplementedError
  end
end
```

And then have `fetch` and `save` (or something similar) defined in each of the adapters.  That'd clean up some of the boilerplate.

Reply:
Agreed, great idea!

Reply:
This is implemented now as well!